### PR TITLE
Refactor Pictures

### PR DESCRIPTION
**Pull request by fmatthew5876**
Depends on #2052 

This refactors the picture data. Now we have one `Game_Pictures` class which manages all the pictures and detaches them from `Game_Screen`. This fixes all the recent picture loading issues #2051 and #2053. It also cleans up initialization and provides some optimizations.

### CODE REVIEW

**Comment by fmatthew5876** [comment]
I did a vtune profile of a game with no pictures. In master it spends 0.4% of time in `Game_Picture::Update()`. With this PR, the update routine doesn't even appear at all in the profile.

**Comment by fmatthew5876** [comment]
Rebased and retested with Frozen Triggers. Also retested my battle pictures test game, along with saving and loading on that map.

This could use a bit more testing to be sure if you guys can help out.

File: src/game_pictures.cpp [hunk 1]
```diff
@@ -255,56 +349,60 @@ void Game_Picture::Move(const MoveParams& params) {
...
 
 	FileRequestAsync* request = AsyncHandler::RequestFile("Picture", name);
 	request->SetGraphicFile(true);
 
-	int pic_id = data.ID;
+	int pic_id = pic.data.ID;
 
 	// FIXME: This lambda is here because it's possible the picture vector can be
 	// resized before the async call of onPictureSpriteReady().
 	// Pictures should be refactored so this ugly hack is not needed.
```
Comment (Ghabry) [inline src/game_pictures.cpp]: Considering that this is now refactored this lambda isn't needed anymore?

Comment (fmatthew5876) [reply src/game_pictures.cpp]: We need to capture the `pic_id` so yes. It could be done other ways, but there is nothing wrong with the lambda really.

**Comment by Ghabry** [comment]
I tested smashbros, this demoscene sequence, Razas (the intro sequence where we already had bugs), "Hentai Film" intro sequence (also had pic bugs years ago) and a few random japanese games. Work for me.
Though for Spritesheet I don't really have any, Frozen Triggers is the best bet.

Save/Load also seems to be crash free.

File: src/game_pictures.cpp (hunk 1 above)
Comment (Ghabry) [reply src/game_pictures.cpp]: Can you remove the FIXME comment at least?

(The other way would be ``request->Bind(&Game_Pictures::OnPictureSpriteReady, this, pic_id)`` as done everywhere else (bet this was added pre-C++11, looks so dated with lambda support))

Comment (fmatthew5876) [reply src/game_pictures.cpp]: Oops, that FIXME should have been removed. Took it out, thanks!

**Pull request merged**